data_management/models.py

```diff
@@ -645,26 +648,44 @@ class ExternalObject(BaseModel):
     `last_updated`: Datetime that this record was last updated
 
     `updated_by`: Reference to the user that updated this record
+
+    `version`: Version identifier of the `DataProduct` associated with this `ExternalObject`
     """
-    ADMIN_LIST_FIELDS = ('doi_or_unique_name', 'title')
+    ADMIN_LIST_FIELDS = ('other_unique_name', 'title', 'version')
 
     data_product = models.ForeignKey(DataProduct, on_delete=models.PROTECT, related_name='external_objects')
-    doi_or_unique_name = models.CharField(max_length=CHAR_FIELD_LENGTH, null=False, blank=False)
+    other_unique_name = models.CharField(max_length=CHAR_FIELD_LENGTH, null=True, blank=True)
+    identifier = models.URLField(max_length=TEXT_FIELD_LENGTH, null=True, unique=True, blank=True)
```

Comment:
I'm not sure what the `unique=` argument does, but presumably either it shouldn't be here, or it should also apply to `other_unique_name`, `title` and `version`?

Reply:
in the identifier definition you mean? It enforces that only one entry in the ExternalObject table can have that identifier. Since it is a URL is this not what we want?

Reply:
No, the uniqueness constraint is on (identifier | other_unique_name) & title & version. So two objects can have the same identifier - for instance it seems that the DOI identifies both a paper and its supplementary materials, so they would both have the same identifier.

Reply:
This isn't a download URL - that's provided by `original_store` - it's an identifier linked to DOIs specifically, like `https://dx.doi.org/10.1%2Fsomething`, which identifies the object as a DOI and then redirects to its landing page.

Reply:
`unique=True` has now been removed in https://github.com/FAIRDataPipeline/data-registry/commit/b966ac01733a4bda0354b5375894ccd36a107adb